### PR TITLE
feat(*): Add support for custom operators

### PR DESCRIPTION
**Pull request by tylerhubert**
Add a new interface that will allow consumers to make Custom Operators and have control over how the
string gets made.

### CODE REVIEW

**Comment by tylerhubert** [comment]
This is the updated version of #9.

File: test/test-operation.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { CustomOperator, quote } from '../src';
+
+export class TestOperation implements CustomOperator {
```
Comment (jdocksey) [inline test/test-operation.ts]: This should be called `TestOperator`.

**Comment by galves** [comment]
Sorry I could only see the "close" button, which I assumed meant "merge and close" but actually it is just that don't have write access to the repo.  Please can you merge yourself.

File: test/test-operation.ts (hunk 1 above)
Comment (tylerhubert) [reply test/test-operation.ts]: Fixed in the next commit, then updated the README to read properly as well.

**Pull request merged**